### PR TITLE
Add LogsDB challenge tests for reindexing

### PR DESCRIPTION
**Pull request by lkts**
This PR adds challenge tests that compare before and after state of the following operations:
* Reindex logsdb data stream into an identical logs data stream
* Reindex logsdb data stream into a standard index mode data stream with the same mapping
* Reindex standard index mode data stream into a logsdb data stream with the same mapping

### CODE REVIEW

**Comment by lkts** [comment]
@elasticmachine update branch

**Review by martijnvg — APPROVED** [review]
LGTM

File: modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/StandardVersusStandardReindexedIntoLogsDbChallengeRestIT.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * This test compares behavior of a standard index mode data stream and a
+ * logsdb data stream containing data reindexed from initial data stream.
+ * There should be no differences between such two data streams.
+ */
+public class StandardVersusStandardReindexedIntoLogsDbChallengeRestIT extends ReindexChallengeRestIT {
```
Comment (kkrik-es) [inline modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/StandardVersusStandardReindexedIntoLogsDbChallengeRestIT.java:21]: Confused, should the name be `StandardReindexedIntoLogsDbChallengeRestIT`?

Comment (kkrik-es) [reply modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/StandardVersusStandardReindexedIntoLogsDbChallengeRestIT.java:21]: Nm, got it.

**Review by kkrik-es — APPROVED** [review]

**Pull request merged**